tests/Microsoft.Identity.Test.Unit/AppTokenProviderTests.cs

```diff
@@ -158,5 +159,60 @@ private AppTokenProviderResult GetAppTokenProviderResult(string differentScopesF
             return token;
         }
 
+        [TestMethod]
+        public async Task MultiThreadedAppTokenProviderCallHasOnlyOneTokenEndpointRequestAsync()
+        {
+            string differentScopesForAt = string.Empty;
+            int totalThreads = 10; // Number of threads to run concurrently
+            int identityProviderHits = 0;
+            int cacheHits = 0;
+            int threadCount = totalThreads;
+
+            var app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                .WithAuthority("https://login.microsoftonline.com/tid")
+                .WithAppTokenProvider((AppTokenProviderParameters parameters) =>
+                {
+                    return Task.FromResult(GetAppTokenProviderResult(differentScopesForAt));
+                })
+                .BuildConcrete();
+
+#pragma warning disable VSTHRD101
```

Comment:
Pragma warning disable should always be followed by #pragma warning enable.

Reply:
removed